cmd/osbuild-jobsite-builder/main.go

```diff
@@ -166,13 +162,8 @@ func (builder *Builder) HandleProvision(w http.ResponseWriter, r *http.Request)
 		0400,
 	)
 
-	defer func() {
-		if cerr := dst.Close(); cerr != nil {
-			err = cerr
```

Comment:
This pattern is potentially cnfusing as a close error would shadow the original error, I think this should check if there is a pre-existing error and only if that is nil override (I know this got removed but still wanted to point it out)

Reply:
Agreed however since the idiom has changed now this part can be discussed in the other comment for the `defer dst.Close()`.